compact_test.go

```diff
@@ -741,3 +744,74 @@ func TestDisableAutoCompactions(t *testing.T) {
 	}
 	testutil.Assert(t, len(db.Blocks()) > 0, "No block was persisted after the set timeout.")
 }
+
+// TestCancelCompactions ensures that when the db is closed
+// any running compaction is cancelled to unblock closing the db.
+func TestCancelCompactions(t *testing.T) {
```

Comment:
Not a fan of this test as I mentioned before, but we can fix it in later PR (sooner than we want - there is high chance it will flake soon I am sure) (:

Reply:
hahaha. DEAL!